core/testsuite/src/test/java/com/blazebit/persistence/testsuite/SizeTransformationTest.java

```diff
@@ -391,7 +392,7 @@ public void work(EntityManager em) {
                 .from(Document.class, "d")
                 .select("SUM(SIZE(d.people))");
 
-        String expected = "SELECT SUM(COALESCE((SELECT COUNT(*) FROM d.people person))) FROM Document d";
+        String expected = "SELECT SUM(COALESCE((SELECT COUNT(*) FROM d.people person),NULL)) FROM Document d";
         assertEquals(expected, cb.getQueryString());
         assertEquals(Long.valueOf(4L), cb.getResultList().get(0));
```

Comment:
Seems the result of a `SUM` is BigDecimal when using EclipseLink: https://travis-ci.org/Blazebit/blaze-persistence/jobs/375353385#L1478
I guess this is legal, you should use `Number` as the return type and do a proper conversion by using `toLong()`